socket.go

```diff
@@ -8,16 +8,30 @@ import (
 )
 
 // parseSockAddr resolves given addr to unix.Sockaddr
-func parseSockAddr(addr string) (unix.Sockaddr, error) {
+func parseSockAddr(addr string) (unix.Sockaddr, int, error) {
 	tAddr, err := net.ResolveTCPAddr("tcp", addr)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
-	var addr4 [4]byte
-	if tAddr.IP != nil {
-		copy(addr4[:], tAddr.IP.To4()) // copy last 4 bytes of slice to array
+
+	if ip := tAddr.IP.To4(); ip != nil {
+		var addr4 [net.IPv4len]byte
+		copy(addr4[:], ip)
+		sAddr := &unix.SockaddrInet4{Port: tAddr.Port, Addr: addr4}
+		return sAddr, unix.AF_INET, nil
+	}
+
+	if ip := tAddr.IP.To16(); ip != nil {
+		var addr16 [net.IPv6len]byte
+		copy(addr16[:], ip)
+		sAddr := &unix.SockaddrInet6{Port: tAddr.Port, Addr: addr16}
+		return sAddr, unix.AF_INET6, nil
+	}
+
+	return nil, 0, &net.AddrError{
+		Err:  "invalid IP address",
```

Comment:
What about embedding the actual address with `unsupported address family`?

Reply:
sgtm, fixed